### PR TITLE
feat(cli): add `qf graph` story visualization command

### PR DESCRIPTION
**Pull request by pvliesdonk**
## Problem
There's no way to visualize the branching story structure after GROW/FILL stages.
Users must manually parse `graph.json` to understand passage connectivity and arc layout.

## Changes
- New `src/questfoundry/visualization.py` module with:
  - `build_story_graph()` — extracts passages, choices, and arcs from the graph
  - `render_dot()` — outputs DOT (Graphviz) format with node shapes by role and colors by arc
  - `render_mermaid()` — outputs Mermaid diagram format
- New `qf graph` CLI command with options:
  - `--format` (`dot`, `mermaid`, `json`)
  - `--output` (file path, stdout default)
  - `--spine-only` (filter to spine arc)
  - `--no-labels` (omit choice labels for cleaner output)
- 27 unit tests covering graph extraction, DOT rendering, Mermaid rendering, and edge cases

## Not Included / Future PRs
- Graphviz rendering (SVG/PNG) — users pipe to `dot` directly
- Subgraph clustering by arc (could add `--cluster-arcs` later)

## Test Plan
- `uv run ruff check src/questfoundry/visualization.py` — passes
- `uv run mypy src/questfoundry/visualization.py` — passes
- `uv run pytest tests/unit/test_visualization.py -x -q` — 27 passed
- Manual: `qf graph -p test-murder-gpt5mini` — produces valid DOT output
- Manual: `qf graph -p test-murder-gpt5mini --format mermaid` — produces valid Mermaid output

## Usage
```bash
qf graph -p myproject | dot -Tsvg -o story.svg   # Render SVG
qf graph -p myproject --format mermaid            # Mermaid output
qf graph -p myproject --spine-only --no-labels    # Clean spine view
```

## Risk / Rollback
- Pure additive change — new module and CLI command, no existing code modified (except CLI wiring)
- No new dependencies

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/questfoundry/visualization.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    for arc_id, arc_type in sorted(arc_names.items()):
+        if arc_type == "spine":
+            color_map[arc_id] = _ARC_COLORS[0]
+    # Then branches
+    idx = 1
+    for arc_id, arc_type in sorted(arc_names.items()):
+        if arc_type != "spine":
+            color_map[arc_id] = _ARC_COLORS[idx % len(_ARC_COLORS)]
+            idx += 1
+    return color_map
```
[the remark this answers is not in the document]
Comment (pvliesdonk) [reply src/questfoundry/visualization.py:294]: Fixed: branch arcs now use `_ARC_COLORS[1:]` slice, so they never collide with the spine color at index 0.

**Comment by pvliesdonk** [comment]
All 5 review comments addressed in the latest push:

1. **Silent fallback** — Now skips choices with missing `from_passage`/`to_passage` and logs a structured warning
2. **Unused list wrapper** — Removed `label_parts`, using `node.label` directly
3. **Arc color bug** — Branch arcs now use `_ARC_COLORS[1:]` slice, preventing spine color collision
4. **CLI format enum** — Using `StrEnum` for the `--format` option; typer handles validation automatically
5. **Mermaid escaping** — Using `&quot;` HTML entity per Mermaid spec
